File: my-credentials-plugin/src/main/java/holygradle/credentials/HideWindowActionListener.java
package holygradle.credentials;

import java.awt.*;
import java.awt.event.*;
 
class HideWindowActionListener implements ActionListener {
    private final Window window;
    private boolean fired = false;
    
    public HideWindowActionListener(Window window) {
        this.window = window;
    }
    
    public void actionPerformed(ActionEvent e) {
        fired = true;
        window.setVisible(false);
    }
    
    public boolean didFire() {
        return fired;
    }
}

File: my-credentials-plugin/src/main/kotlin/holygradle/credentials/HideWindowActionListener.kt
package holygradle.credentials

import java.awt.*
import java.awt.event.*
 
class HideWindowActionListener(private val window: Window) : ActionListener {
    var fired = false
        private set
    
    override fun actionPerformed(e: ActionEvent) {
        fired = true
        window.isVisible = false
    }
}